store/copr/region_cache.go

```diff
@@ -75,59 +75,59 @@ func (l *LocationKeyRanges) splitKeyRangesByBuckets() []*LocationKeyRanges {
 
 	ranges := l.Ranges
 	loc := l.Location
+
+	// Sometimes, there may be some delay between region information and buckets information within the region.
+	// It may cause some gaps between the region startKey and first buckets key and region endKey and the last buckets key.
+	// To easy the implementation of this function, we add region startKey if it is less than the first bucket key and add region endKey if
+	// it is larger than the last bucket key. Without the addition, loc.LocateBucket may return nil which can cause troubles.
+	if len(loc.Buckets.Keys) == 0 || bytes.Compare(loc.StartKey, loc.Buckets.Keys[0]) < 0 {
+		loc.Buckets.Keys = append([][]byte{loc.StartKey}, loc.Buckets.Keys...)
+	}
+
+	if bytes.Compare(loc.Buckets.Keys[len(loc.Buckets.Keys)-1], loc.EndKey) < 0 {
+		loc.Buckets.Keys = append(loc.Buckets.Keys, loc.EndKey)
+	}
+
 	res := []*LocationKeyRanges{}
 	for ranges.Len() > 0 {
 		bucket := loc.LocateBucket(ranges.At(0).StartKey)
-		if bucket == nil {
```

Comment:
Why remove the nil check, seems the `loc.LocateBucket` could still return nil results.